util/ksonnet/ksonnet.go

```diff
@@ -132,12 +139,63 @@ func (k *ksonnetApp) Show(environment string) ([]*unstructured.Unstructured, err
 		if err != nil {
 			return nil, fmt.Errorf("Failed to unmarshal manifest from `ks show`")
 		}
+		err = remarshal(&obj)
+		if err != nil {
+			return nil, err
+		}
 		objs = append(objs, &obj)
 	}
 	// TODO(jessesuen): we need to sort objects based on their dependency order of creation
 	return objs, nil
 }
 
+// remarshal checks resource kind and version and re-marshal using corresponding struct custom marshaller. This ensures that expected resource state is formatter same as actual
+// resource state in kubernetes and allows to find differences between actual and target states more accurate.
+func remarshal(obj *unstructured.Unstructured) error {
+	var newObj interface{}
+	switch obj.GetAPIVersion() + ":" + obj.GetKind() {
+	case "apps/v1beta1:Deployment":
+		newObj = &v1beta1.Deployment{}
+		break
```

Comment:
Style nit: in go, you don't need to stay break after each case.

Reply:
thank you! fixed